lib/route.coffee

```diff
@@ -193,7 +195,8 @@ class share.Route
       endpointContext.user = auth.user
       endpointContext.userId = auth.user._id
       true
-    else false
+    else
+      if typeof auth is 'object' then auth else false
```

Comment:
The problem with this is that if the default login/logout endpoints are disabled (`{defaultAuth: false}`) on an API, and auth is required on an endpoint (`{autRequired: true}`) but not provided by the user, it returns a 200 instead of the expected 401, as `auth` is the object `{userId: undefined, token: undefined}` (so that's returned instead of a falsy value). 

This is causing one of the tests to fail – `Authentication can be required even when the default endpoints aren't configured`. This can only be uncovered by fixing the bcrypt error to run the app by updating accounts-password to v1.3.3. I have a branch `bcrypt-error` with that fix. I'll try to fix this up tomorrow. Let me know if you have any suggestions or preferences on how to handle that case. 

Pinging @rodrigok and @graywolf336 as well.